### PR TITLE
Profile decorator

### PR DESCRIPTION
**Pull request by cguardia**
This was useful to me during recent Karl tests, thought that maybe it could be part of the package.

### CODE REVIEW

**Comment by tseaver** [comment]
Thanks for the patch!  Travis is offline ATM, but I'll bet the failure is due to Python 3.2.  I will remove it on `master` shortly.

**Comment by tseaver** [comment]
Hmm, I see an odd error when testing locally:

```python
======================================================================
ERROR: test_it (repoze.profile.tests.TestProfileDecorator)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/repoze/src/repoze.profile/repoze/profile/tests.py", line 489, in test_it
    do_nothing()
  File "/home/tseaver/projects/repoze/src/repoze.profile/repoze/profile/decorator.py", line 15, in wrapper
    stats.sort_stats(*sort_columns)
  File "/opt/Python-2.6.9/lib/python2.6/pstats.py", line 219, in sort_stats
    sort_tuple = sort_tuple + sort_arg_defs[word][0]
KeyError: 'cumtime'

======================================================================
ERROR: test_limit_lines (repoze.profile.tests.TestProfileDecorator)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/repoze/src/repoze.profile/repoze/profile/tests.py", line 515, in test_limit_lines
    do_nothing()
  File "/home/tseaver/projects/repoze/src/repoze.profile/repoze/profile/decorator.py", line 15, in wrapper
    stats.sort_stats(*sort_columns)
  File "/opt/Python-2.6.9/lib/python2.6/pstats.py", line 219, in sort_stats
    sort_tuple = sort_tuple + sort_arg_defs[word][0]
KeyError: 'cumtime'

----------------------------------------------------------------------
Ran 26 tests in 30.289s
```

It looks like the `cumtime` alias was added in 2.7.  Could you try using `cumulative` instead?

**Comment by cguardia** [comment]
Done.

**Comment by cguardia** [comment]
Did the merge on my branch so it's easy for you to handle here.

**Comment by cguardia** [comment]
Mmm, not sure what to do about those failures.

**Comment by tseaver** [comment]
I don't understand the Py3k failures on Travis.

File: repoze/profile/tests.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import unittest
+import sys
+from StringIO import StringIO
```
Comment (tseaver) [inline repoze/profile/tests.py]: This needs to be `from repoze.profile.compat import StringIO`.

**Comment by tseaver** [comment]
I fixed the Py3k issues, and added coverage for the `if lines == 0:` case.

**Comment by cguardia** [comment]
Thank you!

**Pull request merged**